src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java

```diff
@@ -181,11 +194,47 @@ private void checkParagraphTag(DetailNode tag) {
         else if (newLine == null || tag.getLineNumber() - newLine.getLineNumber() != 1) {
             log(tag.getLineNumber(), MSG_LINE_BEFORE);
         }
+
+        checkIfFollowedByBlockTag(tag);
+
```

Comment:
make it as 
```
if (isFollowedByBlockTag(tag)) {
 log.....
}
```

Reply:
this method logs message with the block tag it found not with `tag` it gets as parameter:

```
                if (blockTags.contains(htmlTagName)) {
                    log(htmlElement.getLineNumber(), htmlElement.getColumnNumber(),
                            MSG_EXTRA_TAG, "<" + htmlTagName + ">");
                }
```

If we want to log as you suggested then we will need to return the block tag and tag name we found from the method. 

The current approach is more suitable then returning a value

Reply:
Ok please return a value, but log should be at checkParagraphTag

Reply:
```
BlockTagViolation blocktag = findParagraphFollowedByBlockTag(tag)
if (blocktag != null) {
 log.....
}
```
and
`class BlockTagViolation (line, column, blockTagName)`

Reply:
Code will be like I suggested above, after fix of https://github.com/checkstyle/checkstyle/pull/15710#discussion_r1782742629

Reply:
done

https://github.com/checkstyle/checkstyle/blob/03f5d5ecffd7969cf64ab60ab12811ee8ce3d1d0/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java#L204-L207